cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -396,11 +397,11 @@ public void upgrade(boolean continueOnFailure) throws Exception {
           continue;
         }
 
-        Location appJarLocation = store.getApplicationArchiveLocation(appId);
+        Location appJarLocation = findAppJarLocation(appId);
         if (appJarLocation == null) {
-          LOG.error(String.format("Unable to get location of jar for app '%s' in namespace '%s'. " +
-              "You will need to re-deploy the app after upgrade.",
-            appId.getNamespaceId(), appId.getId()));
+          // nothing we can do... skip and log a message
+          LOG.warn("Unable to find application jar for app '%s' in namespace '%s'. " +
+            "Please re-deploy the app manually after upgrade.");
```

Comment:
That is an important message. Will this only go to the log? Is there a way to report that this back directly to the UpgradeTool? So that the user who runs the upgrade immediately gets a report of what went wrong?